osu.Game/Screens/OnlinePlay/Components/RoomManager.cs

```diff
@@ -72,9 +72,13 @@ public virtual void JoinRoom(Room room, string? password = null, Action<Room>? o
             currentJoinRoomRequest?.Cancel();
             currentJoinRoomRequest = new JoinRoomRequest(room, password);
 
-            currentJoinRoomRequest.Success += () =>
+            currentJoinRoomRequest.Success += result =>
             {
                 joinedRoom.Value = room;
+
+                AddOrUpdateRoom(result);
+                room.CopyFrom(result); // Also copy back to the source model, since this is likely to have been stored elsewhere.
```

Comment:
This is a char-for-char copy of `CreateRoom()`. I _think_ everything continues to work correctly without the `room.CopyFrom()` line, as opposed to the `CreateRoom()` request which requires it, but I hope to remove this in both cases in the future.  
Just keeping it around to keep things uniform. Can remove if desired.

Reply:
I guess keep it around until the other one can also be removed.